### PR TITLE
Add 'may-spotless-apply' profile

### PR DESCRIPTION
**Pull request by Vlatombe**
I often find myself forgetting to apply spotless before committing.

On some repositories, I have set up a pre-commit git hook running `mvn spotless:check`. However, this has to be set up on each repository that has spotless check enabled, and disabled on those which don't.

I think this goes one step further, by defining a conventional maven profile `may-spotless-apply` that will run `spotless:apply` *only* if spotless is enabled for the given repository.

This allows any user to incorporate `-Pmay-spotless-apply` as part of their usual build command (or defined in their maven settings active profiles section), and have `spotless:apply` automatically called if the project has it enabled.

Example Maven `settings.xml`

```
<settings>
  [...]
  <activeProfiles>
    [...]
    <activeProfile>may-spotless-apply</activeProfile>
  </activeProfiles>
</settings>
```

See also https://github.com/jenkinsci/pom/pull/611

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1461,5 +1461,28 @@
...
+            <artifactId>spotless-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>spotless-apply</id>
+                <goals>
+                  <goal>apply</goal>
+                </goals>
+                <phase>validate</phase>
+                <configuration>
+                  <skip>${spotless.check.skip}</skip>
```
Comment (jglick) [inline pom.xml:1479]: https://github.com/jenkinsci/plugin-pom/blob/64885a6b8ec81f2e474c28a113b850339d4420cb/pom.xml#L126-L127 FTR

**Review by jglick — COMMENTED** [review]
> defined in their maven settings active profiles section

What would this look like? If you just set

```xml
        <profile>
            <id>may-spotless-apply</id>
            <activation>
                <activeByDefault>true</activeByDefault>
            </activation>
        </profile>
```

(with no further configuration in `~/.m2/settings.xml`) does that in fact cause the config in this parent POM to be run?

**Comment by Vlatombe** [comment]
In fact, Maven `settings.xml` has a specific section for profiles you want to activate unconditionally.

```
<settings>
  [...]
  <activeProfiles>
    [...]
    <activeProfile>may-spotless-apply</activeProfile>
  </activeProfiles>
</settings>

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
Worth mentioning in https://github.com/jenkinsci/plugin-pom/blob/master/README.md#formatting I guess.

**Review by timja — COMMENTED** [review]
Can you document it please

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
Thanks for the PR!

File: pom.xml [hunk 2]
```diff
@@ -1461,5 +1461,28 @@
         <maven.compiler.release>21</maven.compiler.release>
       </properties>
     </profile>
+    <profile>
+      <id>may-spotless-apply</id>
+      <build>
```
Comment (jgreffe) [inline pom.xml:1466]: @Vlatombe 
Couldn't the profile be automatically applied depending on a property?

```
      <id>may-spotless-apply</id>
      <activation>
        <property>
          <name>!spotless.check.skip</name>
        </property>
      </activation>
      <build>
```

Comment (jglick) [reply pom.xml:1466]: IIRC you cannot activate profiles based on properties defined in the project POM, only on user properties.

**Pull request merged**